internal/provider/resources/install/okta/directory_listing_staged.go

```diff
@@ -0,0 +1,157 @@
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/p0-security/terraform-provider-p0/internal"
+	installresources "github.com/p0-security/terraform-provider-p0/internal/provider/resources/install"
+)
+
+var _ resource.Resource = &OktaDirectoryListingStaged{}
+var _ resource.ResourceWithImportState = &OktaDirectoryListingStaged{}
+var _ resource.ResourceWithConfigure = &OktaDirectoryListingStaged{}
+
+func NewOktaDirectoryListingStaged() resource.Resource {
+	return &OktaDirectoryListingStaged{}
+}
+
+type OktaDirectoryListingStaged struct {
+	installer *installresources.Install
+}
+
+type oktaDirectoryListingStagedModel struct {
+	Domain string       `tfsdk:"domain"`
+	Jwk    types.Object `tfsdk:"jwk"`
+}
+
+type oktaDirectoryListingStagedJson struct {
+	KeyId     string `json:"keyId"`
+	PublicKey string `json:"publicKey"`
+	State     string `json:"state"`
+}
+
+type oktaDirectoryListingStagedApi struct {
+	Item oktaDirectoryListingStagedJson `json:"item"`
+}
+
+func (r *OktaDirectoryListingStaged) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_okta_directory_listing_staged"
+}
+
+func (r *OktaDirectoryListingStaged) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: `A staged installation of P0, on an Okta organization for directory listing.
+
+For instructions on using this resource, see the documentation for ` + "`p0_okta_directory_listing`.",
+		Attributes: map[string]schema.Attribute{
+			"domain": schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "The domain of the Okta organization",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"jwk": schema.ObjectAttribute{
+				Computed:            true,
+				MarkdownDescription: "The public JSON Web Key (JWK) added to the Okta App",
+				AttributeTypes: map[string]attr.Type{
+					"kty": types.StringType,
+					"kid": types.StringType,
+					"e":   types.StringType,
+					"n":   types.StringType,
+				},
+			},
+		},
+	}
+}
+
+func (r *OktaDirectoryListingStaged) getId(data any) *string {
+	model, ok := data.(*oktaDirectoryListingStagedModel)
+	if !ok {
+		return nil
+	}
+	return &model.Domain
+}
+
+func (r *OktaDirectoryListingStaged) getItemJson(json any) any {
+	return json
+}
+
+func (r *OktaDirectoryListingStaged) fromJson(ctx context.Context, diags *diag.Diagnostics, id string, jsonData any) any {
+	data := oktaDirectoryListingStagedModel{}
+	jsonv, ok := jsonData.(*oktaDirectoryListingStagedApi)
+	if !ok {
+		return nil
+	}
+
+	data.Domain = id
+
+	var jwk Jwk
+	if err := json.Unmarshal([]byte(jsonv.Item.PublicKey), &jwk); err != nil {
+		diags.AddError("Error parsing JWK", err.Error())
+		return nil
+	}
+
+	jwkAttr, jwkDiags := types.ObjectValueFrom(ctx, map[string]attr.Type{
```

Comment:
suggestion: put this logic in a shared function then call it from both the staged and final resources